src/main/java/org/jitsi/jigasi/JvbConference.java

```diff
@@ -760,6 +766,24 @@ private void discoverComponentAddresses()
                     logger.info(String.format("%s Discovered %s for %oms.",
                         this.callContext, avModerationAddress, System.currentTimeMillis() - startQuery));
                 }
+
+                DiscoverInfo.Identity roomMetadataIdentity = info.getIdentities().stream().
+                        filter(di -> di.getCategory().equals("component") && di.getType().equals("room_metadata"))
+                        .findFirst().orElse(null);
+
+                // we process room metadata messages only when we are transcribing
+                if (roomMetadataIdentity != null && this.gatewaySession instanceof TranscriptionGatewaySession)
+                {
+                    if (roomMetadataListener != null)
```

Comment:
Any reason not to make the field `final` and always initialize it? You can remove these 5 lines of code and not worry about this executing after a re-connect for example